androidApp/src/main/java/com/pulselink/auth/FirebaseAuthManager.kt

```
@@ -53,11 +56,17 @@ class FirebaseAuthManager @Inject constructor(
 
-    suspend fun signInSmsOnly(): Result<FirebaseUser> {
+    suspend fun signInSmsOnly(): Result<Unit> {
         return runCatching {
-            auth.signInAnonymously().await().user ?: error("Anonymous sign-in returned no user")
-        }.onFailure { error ->
-            Log.w(TAG, "SMS-only sign-in failed", error)
+            try {
+                auth.signInAnonymously().await()
+                isOfflineMode = false
+                // updateAuthState will be called by listener
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)